tests/main/validate-container-happy/task.yaml

```
@@ -0,0 +1,40 @@
+summary: check the symlinks following the right track
+
+environment:
+    SNAP: test-snapd-validate-container-happy
+    SNAP_INSTALL_DIR: $(pwd)/snap-install-dir
+    SNAP_MOUNT_DIR: $(os.paths snap-mount-dir)
+
+prepare: |
+    snap pack test-snapd-validate-container-happy
+
+execute: |
+
+    # We shouldn't use relative symlinks in Github as they cannot be packed correctly.
+    # So here let's test whether we can still pack such symlinks within a snap and use if needed.
+    # First we "try" to unpack the snap structure and untar the symlinks
+    # Then we pack the snap with these symlinks and then install
+    # Finally we check to see if the symlinks actually support the intervined symlinks
+
+    # Untar the symlinks 
+    tar -xvf "$SNAP"/hell/hell.tar -C "$SNAP"/hell
+
+    snap try "$SNAP"
+    # Check to see if the symlinks exist
+    ls -l "$SNAP"/hell | MATCH "foo -> baz -> qux"
+    ls -l "$SNAP"/hell | MATCH "foo -> bar -> qux"
+    ls -l "$SNAP"/hell | MATCH "bar -> baz -> qux"
+    ls -l "$SNAP"/hell | MATCH "foo -> bar -> baz"
```

these checks should be for $SNAP_MOUNT_DIR 

```suggestion
    ls -l "$SNAP_MOUNT_DIR"/hell | MATCH "foo -> baz -> qux"
    ls -l "$SNAP_MOUNT_DIR"/hell | MATCH "foo -> bar -> qux"
    ls -l "$SNAP_MOUNT_DIR"/hell | MATCH "bar -> baz -> qux"
    ls -l "$SNAP_MOUNT_DIR"/hell | MATCH "foo -> bar -> baz"
```

---

also I have a slight preference for checking the symlinks with readlink instead of `ls -l`

---

I prefer readlink myself too, just wanted to diversify the checks. Switched the checks during the "try" to readlink as well.